src/current_plugin.cpp

```diff
@@ -1,46 +1,42 @@
 
 #include "extism.hpp"
+#include <cstring>
 
 namespace extism {
 
-uint8_t *CurrentPlugin::memory() {
+uint8_t *CurrentPlugin::memory() const {
   return extism_current_plugin_memory(this->pointer);
 }
-uint8_t *CurrentPlugin::memory(MemoryHandle offs) {
+uint8_t *CurrentPlugin::memory(MemoryHandle offs) const {
   return this->memory() + offs;
 }
 
-ExtismSize CurrentPlugin::memoryLength(MemoryHandle offs) {
+ExtismSize CurrentPlugin::memoryLength(MemoryHandle offs) const {
   return extism_current_plugin_memory_length(this->pointer, offs);
 }
 
-MemoryHandle CurrentPlugin::memoryAlloc(ExtismSize size) {
+MemoryHandle CurrentPlugin::memoryAlloc(ExtismSize size) const {
   return extism_current_plugin_memory_alloc(this->pointer, size);
 }
 
-void CurrentPlugin::memoryFree(MemoryHandle handle) {
+void CurrentPlugin::memoryFree(MemoryHandle handle) const {
   extism_current_plugin_memory_free(this->pointer, handle);
 }
 
-void CurrentPlugin::output(const std::string &s, size_t index) {
+void CurrentPlugin::output(const std::string &s, size_t index) const {
   this->output((const uint8_t *)s.c_str(), s.size(), index);
 }
 
-void CurrentPlugin::output(const uint8_t *bytes, size_t len, size_t index) {
-  if (index < this->nInputs) {
+void CurrentPlugin::output(const uint8_t *bytes, size_t len,
+                           size_t index) const {
+  if (index < this->nOutputs) {
```

Comment:
Should we do something if the index is out of bounds?

Reply:
The output functions now return a `bool`.

Reply:
Nice!